lib/shopify_app/managers/webhooks_manager.rb

```diff
@@ -41,21 +41,23 @@ def add_registrations
           ShopifyAPI::Webhooks::Registry.add_registration(
             topic: attributes[:topic],
             delivery_method: attributes[:delivery_method] || :http,
-            path: attributes[:address],
-            handler: webhook_job_klass(attributes[:topic]),
+            path: attributes[:path] || attributes[:address],
+            handler: webhook_job_klass(attributes[:path]),
             fields: attributes[:fields]
           )
         end
       end
 
       private
 
-      def webhook_job_klass(topic)
-        webhook_job_klass_name(topic).safe_constantize || raise(ShopifyApp::MissingWebhookJobError)
+      def webhook_job_klass(path)
+        webhook_job_klass_name(path).safe_constantize || raise(ShopifyApp::MissingWebhookJobError)
       end
 
-      def webhook_job_klass_name(topic)
-        [ShopifyApp.configuration.webhook_jobs_namespace, "#{topic.gsub("/", "_")}_job"].compact.join("/").classify
+      def webhook_job_klass_name(path)
+        job_file_name = path.split("/").last
+        [ShopifyApp.configuration.webhook_jobs_namespace,
+         "#{job_file_name.gsub("/", "_")}_job",].compact.join("/").classify
```

Comment:
I'm a little confused how `job_file_name` could contain a `"/"` here?

Reply:
ah, you are right! This bit was rolled forward to the old implementaiton that used the topic instead of path to generate this name.